apps/checkout/src/lib/utils/address.ts

```diff
@@ -0,0 +1,24 @@
+import { AddressField } from "@lib/globalTypes";
+
+const addressFieldsOrder: AddressField[] = [
+  "name",
+  "companyName",
+  "streetAddress1",
+  "streetAddress2",
+  "city",
+  "postalCode",
+];
+
+// api doesn't order the fields but we want to
+export const getSortedAddressFields = (addressFields: AddressField[] = []) =>
+  addressFieldsOrder.reduce((result, orderedAddressField) => {
+    if (!addressFields.includes(orderedAddressField)) {
+      return result;
+    }
+
+    return [...result, orderedAddressField];
+  }, [] as AddressField[]);
+
+export const getSortedAddressFieldsFromAddress = (
+  address: Partial<Record<AddressField, any>>
+) => getSortedAddressFields(Object.keys(address) as AddressField[]);
```

Comment:
The implementation would be a bit cleaner if we didn't map it to an array as the middle step, if `getSortedAddressFields` isn't useful on it's own I'd skip it and make it a single function:
```
export const getSortedAddressFields = (address: Partial<Record<AddressField, any>>) =>
  addressFieldsOrder.reduce((result, orderedAddressField) => {
    if (orderedAddressField in address) {
      return result;
    }

    return [...result, orderedAddressField];
  }, [] as AddressField[]);
```

Reply:
I think `getSortedAddressFields` will be used on it's own once we add it to the address form, where we get address fields as string[] straight from the api